sdx-submitter.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/streadway/amqp"
+	"io/ioutil"
+	"os"
+)
+
+func main() {
+	// Read a source file and place it on a rabbit topic exchange
+	// Exchange , Queue and binding must be in place before use
+
+	var url, exchange, routingKey, encryptionKey, signingKey, name, password string
+	var port, host, queue, vhost, messageFilePath string
+	var msgBody []byte
+	var err error
+	const PRINT_MSG_CHAR_COUNT = 20 // only print the first few characters of the message
+
+	// access command line parameters
+
+	flag.StringVar(&name, "n", "", "name of the rabbit user")
+	flag.StringVar(&password, "p", "", "password of the rabbit user")
+	flag.StringVar(&port, "rport", "", "port used to connect to rabbit")
+	flag.StringVar(&host, "rhost", "", "hostname used to connect to rabbit")
+	flag.StringVar(&vhost, "v", "", "vhostname used to connect to rabbit")
+	flag.StringVar(&url, "u", "", "url connection string ")
+	flag.StringVar(&queue, "q", "", "name of the rabbit queue")
+	flag.StringVar(&exchange, "x", "", "name of the rabbit exchange")
+	flag.StringVar(&routingKey, "r", "", "rabbit routing key")
+	flag.StringVar(&encryptionKey, "e", "", "path to a private key file used for encryption")
+	flag.StringVar(&signingKey, "s", "", "path to a private key used for signing")
+	flag.StringVar(&messageFilePath, "f", "", "path to filename to send")
```

Comment:
Try to strip down the args to the ones needed for a particular publish (pretty much just the file name or list of files to publish probably).

Can get the rest of the config from a local configuration file that would be put into place to load from when the script is deployed.

Thinking about it I wouldn't look to get any of the vars from the environment - this is going to be deployed/run on demand rather than as an installed and running service (even from a local laptop maybe). Keeping things in a file (or two - like with manifests for cloud foundry) would keep things neat - the env vars used by other running services wouldn't be available to piggy back on and would have to be manually loaded anyhow.

Reply:
The only mandatory parameter is the file name (and the two key files when we use them ) , all others are defaulted . Will change to use a config file instead of environment variables . So the priority will be command line , config file , default values for those params that can be defaulted . File name , and later on the signing and encryption key files we can only take from the command line  I  think. Is it possible to merge what we have so that Gemma can get on with setting up Travis please ? I can do this change on a new branch ?